spec/models/project_spec.rb

```diff
@@ -257,6 +260,7 @@
 
           expect(project.status).to eq("Deprecated")
           expect(project.deprecation_reason).not_to eq(nil)
+          expect(project.status_checked_at).to eq(Time.current)
```

Comment:
Should these also assert that the updated_at was not changed, as that is the specific behavior we're looking for?

Reply:
Good catch, I updated the specs.

Just to check, though, should the `updated_at` be changed when the `status` field does actually change? I assumed that we want `updated_at` to change when we make an actual change to value of `status`, but just not change `updated_at` only when we do a check. See behavior of specs.

If my assumption is incorrect, it's an easy fix; we just need to change the `update_attribute` calls for `status` to be `update_column` instead.

Reply:
I think the correct behavior is that if `status_checked_at` is the only value changing, then we don't update the `updated_at`. All other behavior should remain the same, e.g. if another column is changed, then `updated_at` should change as well. `status_checked_at` should be different from `updated_at`, because we want to be able to check the status regularly, even if the status/other columns don't actually change. cc @havocp for additional thoughts.

Reply:
I think that's right - updated_at will cause the project to come back as changed, when it actually hasn't. I would test that when we set `status_checked_at` _and the status is unchanged_ then we don't modify `updated_at`, but if the actual status or deprecation reason changes, then we should modify `updated_at`.

Reply:
Cool, given that then the behavior is as expected!

Reply:
Cool, given that then the behavior is as expected!